src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,30 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor(popupSelector) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector("#confirm-delete-form");
+  }
+
+  setSubmitAction(handleFormSubmit) {
+    this._handleFormSubmit = handleFormSubmit;
+  }
+
+  setEventListeners() {
+    this._popupForm.addEventListener("submit", (event) => {
+      event.preventDefault();
+      this._handleFormSubmit();
+    });
+    super.setEventListeners();
+  }
+
+  loadingButtonState() {
+    this._confirmBtn = this._popupElement.querySelector("#delete-card-btn");
+
+    this._confirmBtn.textContent = "Deleting...";
+  }
+
+  resetButtonState() {
+    this._confirmBtn.textContent = "Yes";
```

Comment:
`this._confirmBtn` doesn't exist in your class. If you call `resetButtonState` before `loadingButtonState` then your app will crash

So, the best variant is to find the button in line 6 only once